// Copyright 2017 Intel Corporation
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
// -----------------------------------------------------------------------------

syntax = "proto3";

option java_multiple_files = true;
option java_package = "sawtooth.identity.protobuf";

message IdentityPayload {
    enum IdentityType {
      IDENTITY_TYPE_UNSET = 0;
      POLICY = 1;
      ROLE = 2;
    }

    // What type of payload this is for
    IdentityType type = 1;

    // Serialize version of a role or policy
    bytes data = 2;

}
